statics/js/components/graph-layout.js

```diff
@@ -1,6 +1,173 @@
+      const deltaMillis = metrics.Last - metrics.Start;
+      const elapsedMillis = Date.now() - new Date(metrics.Last);
+
+      if (deltaMillis === 0) {
+        return 0;
+      }
+      if (elapsedMillis > maxClockSkewMillis) {
+        return 0;
+      }
+
+      return Math.floor(8 * totalByte * 1000 / deltaMillis); // bits-per-second 
+    },
+
+    setup: function(topology) {
+      this.topology = topology;
+    },
+
+    updateData: function(link) {
+      if (!link.target.metadata.LastUpdateMetric) {
+        return;
+      }
+
+      const defaultBandwidthBaseline = 1024 * 1024 * 1024; // 1 gbps
+
+      link.bandwidthBaseline = (this.topology.bandwidth.bandwidthThreshold === 'relative') ?
+        link.target.metadata.Speed || defaultBandwidthBaseline : 1;
+
+      link.bandwidthAbsolute = this.bandwidthFromMetrics(link.target.metadata.LastUpdateMetric);
+      link.bandwidth = link.bandwidthAbsolute / link.bandwidthBaseline;
+    },
+ 
+    hasData: function(link) {
+      if (!link.target.metadata.LastUpdateMetric) {
+        return false;
+      }
+
+      if (!link.bandwidth) {
+        return false;
+      }
+      return link.bandwidth > this.topology.bandwidth.active;
+    },
+
+    getText: function(link) {
+      return bandwidthToString(link.bandwidthAbsolute);
+    },
+
+    isActive: function(link) {
+      return (link.bandwidth > this.topology.bandwidth.active) && (link.bandwidth < this.topology.bandwidth.warning);
+    },
+
+    isWarning: function(link) {
+      return (link.bandwidth >= this.topology.bandwidth.warning) && (link.bandwidth < this.topology.bandwidth.alert);
+    },
+
+    isAlert: function(link) {
+      return link.bandwidth >= this.topology.bandwidth.alert;
+    },
+  },
+});
+
+var LinkLabelLatency = Vue.extend({
+  mixins: [apiMixin],
+
+  methods: {
+    setup: function(topology) {
+      this.active = 0;
+      this.warning = 10
+      this.alert = 100;
+    },
+
+    updateLatency: function(link, a, b) {
+      link.latencyTimestamp = Math.max(a.Last, b.Last);
+      link.latency = Math.abs(a.RTT - b.RTT) / 1000000;
+    },
+
+    flowQuery: function(nodeTID, trackingID, limit) {
+      let query = `G.Flows()`;
+      query += `.Has("NodeTID", "${nodeTID}")`;
+      if (typeof trackingID !== 'undefined') {
+        query += `.Has("TrackingID", "${trackingID}")`;
+      }
+      query += `.Has("RTT", NE(0))`;
+      query += `.Sort().Limit(${limit})`;
+      return this.$topologyQuery(query)
+    },
+
+    updateData: function(link) {
+      var self = this;
+
+      const a = link.source.metadata;
+      const b = link.target.metadata;
+
+      if (!a.Capture) {
+        return;
+      }
+      if (!b.Capture) {
+        return;
+      }
+
+      this.flowQuery(a.TID, undefined, 10)
+        .then(function(flows) {
+          for (let i in flows) {
+            const aFlow = flows[i];
+            self.flowQuery(b.TID, aFlow.TrackingID, 1)
```

Comment:
for readability would be better to have 2 distinct query, like flowQueryNodeTID() and flowQueryTrackingID()

Reply:
ok, will fix